src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
Well the current IDE driver is not based on compilation events. Once John finishes his prototype, we can get rid of the IDE driver, and hence this API, completely. Till then, we need this workaround.